packages/adp-tooling/src/types.ts

```diff
@@ -199,6 +227,29 @@ export const enum ChangeType {
     CHANGE_INBOUND = 'appdescr_app_changeInbound'
 }
 
+/**
+ * Enumerates the types of projects that can be made, each representing a specific kind of project structure.
+ */
+export const enum ProjectType {
+    ON_PREM = 'OnPrem',
+    S4 = 'S4',
+    CF = 'CF'
+}
+
+/**
+ * Maps a ChangeType to the corresponding data structure needed for that type of change.
+ * This conditional type ensures type safety by linking each change type with its relevant data model.
+ *
+ * @template T - A subtype of ChangeType indicating the specific type of change.
+ */
+export type ProjectGeneratorData<T extends ProjectType> = T extends ProjectType.ON_PREM
```

Comment:
It would be good to remove the generator key word from the interface (if it is used by the create module there you are not using the generator)